lib/irb.rb

```diff
@@ -239,6 +240,7 @@ def read_input(prompt)
     end
 
     def readmultiline
+      @prompt_part_cache = {}
```

Comment:
I think we can create a wrapper to ensure wherever we can generate prompt we have this set and cleaned. Like

```rb
with_prompt_part_cached do
  ...
end
```